File: OPENAPI-REST-API/openapi-client/scala-gatling/src/gatling/scala/org/openapitools/client/model/DomainBillingExtra.scala

package org.openapitools.client.model


case class DomainBillingExtra (
    _order: Option[DomainOrderResponse],
    _order_id: Option[String],
    _domain_id: Option[String],
    _provProcessPending: Option[DomainProvProcessPending],
    _email: Option[String],
    _firstname: Option[String],
    _lastname: Option[String],
    _company: Option[String],
    _address: Option[String],
    _address2: Option[String],
    _address3: Option[String],
    _city: Option[String],
    _state: Option[String],
    _zip: Option[String],
    _country: Option[String],
    _phone: Option[String],
    _fax: Option[String]
)
object DomainBillingExtra {
    def toStringBody(var_order: Object, var_order_id: Object, var_domain_id: Object, var_provProcessPending: Object, var_email: Object, var_firstname: Object, var_lastname: Object, var_company: Object, var_address: Object, var_address2: Object, var_address3: Object, var_city: Object, var_state: Object, var_zip: Object, var_country: Object, var_phone: Object, var_fax: Object) =
        s"""
        | {
        | "order":$var_order,"order_id":$var_order_id,"domain_id":$var_domain_id,"provProcessPending":$var_provProcessPending,"email":$var_email,"firstname":$var_firstname,"lastname":$var_lastname,"company":$var_company,"address":$var_address,"address2":$var_address2,"address3":$var_address3,"city":$var_city,"state":$var_state,"zip":$var_zip,"country":$var_country,"phone":$var_phone,"fax":$var_fax
        | }
        """.stripMargin
}

File: OPENAPI-REST-API/openapi-client/zapier/models/DomainBillingExtra.js
const utils = require('../utils/utils');
const DomainOrderResponse = require('../models/DomainOrderResponse');
const DomainProvProcessPending = require('../models/DomainProvProcessPending');

module.exports = {
    fields: (prefix = '', isInput = true, isArrayChild = false) => {
        const {keyPrefix, labelPrefix} = utils.buildKeyAndLabel(prefix, isInput, isArrayChild)
        return [
            ...DomainOrderResponse.fields(`${keyPrefix}order`, isInput),
            {
                key: `${keyPrefix}order_id`,
                label: `[${labelPrefix}order_id]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}domain_id`,
                label: `[${labelPrefix}domain_id]`,
                type: 'string',
            },
            ...DomainProvProcessPending.fields(`${keyPrefix}provProcessPending`, isInput),
            {
                key: `${keyPrefix}email`,
                label: `[${labelPrefix}email]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}firstname`,
                label: `[${labelPrefix}firstname]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}lastname`,
                label: `[${labelPrefix}lastname]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}company`,
                label: `[${labelPrefix}company]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}address`,
                label: `[${labelPrefix}address]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}address2`,
                label: `[${labelPrefix}address2]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}address3`,
                label: `[${labelPrefix}address3]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}city`,
                label: `[${labelPrefix}city]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}state`,
                label: `[${labelPrefix}state]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}zip`,
                label: `[${labelPrefix}zip]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}country`,
                label: `[${labelPrefix}country]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}phone`,
                label: `[${labelPrefix}phone]`,
                type: 'string',
            },
            {
                key: `${keyPrefix}fax`,
                label: `[${labelPrefix}fax]`,
                type: 'string',
            },
        ]
    },
    mapping: (bundle, prefix = '') => {
        const {keyPrefix} = utils.buildKeyAndLabel(prefix)
        return {
            'order': utils.removeIfEmpty(DomainOrderResponse.mapping(bundle, `${keyPrefix}order`)),
            'order_id': bundle.inputData?.[`${keyPrefix}order_id`],
            'domain_id': bundle.inputData?.[`${keyPrefix}domain_id`],
            'provProcessPending': utils.removeIfEmpty(DomainProvProcessPending.mapping(bundle, `${keyPrefix}provProcessPending`)),
            'email': bundle.inputData?.[`${keyPrefix}email`],
            'firstname': bundle.inputData?.[`${keyPrefix}firstname`],
            'lastname': bundle.inputData?.[`${keyPrefix}lastname`],
            'company': bundle.inputData?.[`${keyPrefix}company`],
            'address': bundle.inputData?.[`${keyPrefix}address`],
            'address2': bundle.inputData?.[`${keyPrefix}address2`],
            'address3': bundle.inputData?.[`${keyPrefix}address3`],
            'city': bundle.inputData?.[`${keyPrefix}city`],
            'state': bundle.inputData?.[`${keyPrefix}state`],
            'zip': bundle.inputData?.[`${keyPrefix}zip`],
            'country': bundle.inputData?.[`${keyPrefix}country`],
            'phone': bundle.inputData?.[`${keyPrefix}phone`],
            'fax': bundle.inputData?.[`${keyPrefix}fax`],
        }
    },
}
